### ISSUE TITLE
non-deterministic behavior with batch size > 1

### ISSUE THREAD

**Issue by FiorenzoParascandolo1** [yuwenmichael/Grounding-DINO-Batch-Inference#4]
I noticed this strange behavior: the logits outputs of the model vary slightly with respect to different runs, resulting in non-deterministic behavior. This effect does not happen in case of batch size = 1. Do you know what the reason could be? The difference between the logits (comparing two separate script executions) increases as the batch size increases
